service/src/main/java/bio/terra/cbas/dependencies/sam/BearerTokenFilter.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.cbas.dependencies.sam;
+
+import java.io.IOException;
+import java.util.Optional;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import org.springframework.http.HttpHeaders;
+import org.springframework.stereotype.Component;
+import org.springframework.web.context.request.RequestAttributes;
+import org.springframework.web.context.request.RequestContextHolder;
+
+/**
+ * <a href="https://docs.oracle.com/javaee/6/api/javax/servlet/Filter.html">Servlet filter</a> that
+ * inspects the incoming request for an Authorization: Bearer ... token, and saves any token it
+ * finds into the current thread-local RequestContextHolder.
+ *
+ * <p>Filters are contructed once (hence the @Component tag), while doFilter is called for each
+ * request/response pair. Read the Javadoc for `doFilter` to learn more about its function. By
+ * isolating the request to this class, we can access thread-specific RequestContextHolders without
+ * worrying about whether a given `request` object is up-to-date. The filter is necessary to inspect
+ * certain request properties like headers, while the RequestContextHolders serves more as a
+ * dictionary for specific keyed attributes.
+ *
+ * <p>Note that this filter does not validate or inspect the token; it just extracts it from the
+ * request, allowing it to be sent as-is from CBAS to other services such as Sam.
+ *
+ * <p>Adapted from <a
+ * href="https://github.com/DataBiosphere/terra-workspace-data-service/blob/b479672d36545d0b45766b3e0e09564d077a19b8/service/src/main/java/org/databiosphere/workspacedataservice/sam/BearerTokenFilter.java">WDS</a>
+ */
+@Component
+public class BearerTokenFilter implements Filter {
+  public static final String ATTRIBUTE_NAME_TOKEN = "bearer-token-attribute";
+  private static final String BEARER_PREFIX = "Bearer ";
+
+  @Override
+  public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain)
```

Comment:
I hadn't seen this before so this approach is new to me and I'm not sure of the pros/cons vs the approach I took with adding support for `BearerToken`. This code uses a filter to populate the request context but then don't make use of using the context to directly inject a request scoped bean, which seems like a missed opportunity to me.

But I think a bigger issue is that you don't need a filter to do this, which makes things more complicated. You can instead define a custom request scoped bean which produces `BearerToken` and inject it directly where needed.

The full PR is here https://github.com/DataBiosphere/terra-data-catalog/pull/37

But the basic idea is to add a bean generator to your `Configuration`:
```
  @Bean
  @RequestScope
  public BearerToken bearerToken(HttpServletRequest request) {
    return new BearerTokenFactory().from(request);
  }
```

where `BearerTokenFactory` is from terra-common-lib. Once you've done that, you can inject the bearer token directly in your code, like this:
```
  @Autowired
  public SamService(SamConfiguration samConfig, SamClient samClient, BearerToken bearerToken) {
    this.samConfig = samConfig;
    this.samClient = samClient;
    this.bearerToken = bearerToken;
  }
```

Note that since `BearerToken` is a request scoped bean, each request gets is own copy even though `SamService` is a singleton bean.